Add new skins

Codidodido opened the pull request:
- Add Luigi & Rex skins
- Display player skin in its slider

Codidodido committed:
Add Luigi sprites

Codidodido committed:
Add Rex sprites

Codidodido committed:
Create Luigi animations

Codidodido committed:
Create Rex animations

Codidodido committed:
Add Luigi, Mario and Rex prefab to Resources folder

Codidodido committed:
Modify skins scriptable object and skin asset

- Rename skinSprite to skinIcon
- Add skinPrefab [save skin prefab which is located in resources folder]

Codidodido committed:
Modify skin card controller, displayer and skin preview navigator script

- Add private skin prefab property to skin card controller and set it with `SetCard()` method by skin card displayer
- Save skin prefab name instead of card skin sprite name into player prefs `skin` key
- Display player prefs skin or mario if it is not set before by default with skin preview navigator

Codidodido committed:
Attach skin asset to skin shop's skin displayer

Codidodido committed:
Edit skin shop button in menu

- Load skin shop scene by pressing skin shop button

Codidodido committed:
Create skin script

- Display player prefs skin with 'skin' script

Codidodido committed:
Modify player prefab

Attach skin script to player prefab

Codidodido committed:
Modify skin & skin preview navigator script

- Check if player prefs has key instead of null check

Codidodido committed:
Edit skin shop scene

- Load menu scene by pressing menu button

Codidodido committed:
Sync slider

- Display player skin in its slider

URAmiRBin commented on Assets/Scripts/PositionMapper.cs:
```diff
@@ -30,5 +30,6 @@ private void CheckPlayerPosition()
     public void SetTarget(GameObject targetObject)
     {
         target = targetObject;
+        slider.handleRect.GetComponent<Image>().sprite = targetObject.GetComponent<SpriteRenderer>().sprite;
```
this line is too complicated, make it more readable.

URAmiRBin commented on Assets/Scripts/Skin.cs:
```diff
@@ -0,0 +1,25 @@
+        }
+        else
+        {
+            _animator.runtimeAnimatorController = Resources.Load<GameObject>("Mario_Prefab").GetComponent<Animator>().runtimeAnimatorController;
```
this like is also too complicated, if you only want the animator, why don't you just load the animator itself, loading the whole prefab does not look optimal

Codidodido committed:
Apply pull request review suggestions

- Refactor position mapper script
|- Add `UpdateGraphics` method
- Refactor scripts related to skin
- Split characters prefabs in resources into their sprite and animator controller
